Gems/AWSGameLift/cdk/aws_gamelift/flex_match/matchmaking.py

```diff
@@ -0,0 +1,67 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+import typing
+
+from aws_cdk import (
+    core,
+    aws_gamelift as gamelift
+)
+
+from . import flex_match_configurations
+
+BACKFILL_MODE_MANUAL = 'MANUAL'
+FLEX_MATCH_MODE_STANDALONE = 'STANDALONE'
+
+class Matchmaking:
+    """
+    Create a matchmaking rule set and matchmaking configuration for Gamelift FlexMatch.
+    For more information about Gamelift FlexMatch, please check
+    https://docs.aws.amazon.com/gamelift/latest/flexmatchguide/match-intro.html
+    """
+    def __init__(self, stack: core.Stack, game_session_queue_arns: typing.List[str]):
+        rule_set = gamelift.CfnMatchmakingRuleSet(
+            scope=stack,
+            id='MatchmakingRuleSet',
+            name=f'{stack.stack_name}-MatchmakingRuleSet',
+            rule_set_body=flex_match_configurations.RULE_SET_BODY
+        )
+
+        # AUTOMATIC FlextMatch mode is not available when there's no game session queue.
+        flex_match_mode = FLEX_MATCH_MODE_STANDALONE if len(game_session_queue_arns) == 0 \
+            else flex_match_configurations.FLEX_MATCH_MODE

```

Comment:
I don't think we should provide standalone flexmatch, as within O3DE, it is designed within AWSGameLift gem together.

Reply:
Will do.